patroni/dcs/kubernetes.py

```diff
@@ -816,6 +816,10 @@ def client_path(self, path: str) -> str:
     def leader_path(self) -> str:
         return super(Kubernetes, self).leader_path[:-7 if self._api.use_endpoints else None]
 
+    @property
+    def should_create_leader_annotation_paths(self) -> Set[str]:
+        return {self.config_path, self.sync_path, self.failover_path}
+
```

Comment:
I don't think we need it, because effectively we want to protect all endpoints created by Patroni.

Reply:
Since endpoints were planned to be deprecated a long time ago, the reason why current endpoints can still work is that there is an endpointslicemirroring_controller mirroring a truly working endpointslice for it.

If we add this annotation to leader endpoints, then endpointslicemirroring_controller will skip it thus it won't accept traffic anymore, users have to define its own ingress service with label selector like role=primary. this seems breaked current behavior.